src/Microsoft.Diagnostics.Monitoring.HostingStartup/ParameterCapturing/FunctionProbes/FunctionProbesManager.cs

```diff
@@ -277,7 +282,7 @@ public async Task StartCapturingAsync(IList<MethodInfo> methods, CancellationTok
             }
             catch
             {
-                FunctionProbesStub.InstrumentedMethodCache = null;
+                FunctionProbesStub.Cache = null;
```

Comment:
Did we want something more deterministic here? For example, call Clear() first and then remove the reference?